fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -1951,31 +1951,47 @@ class DDTeamCollectionImpl {
 		return Void();
 	}
 
+	ACTOR static Future<Void> perpetualStorageWiggleRest(DDTeamCollection* self) {
+		state bool takeRest = true;
+		state Promise<int64_t> avgShardBytes;
+		while (takeRest) {
+			// avoid busy spinning
+			wait(delay(SERVER_KNOBS->PERPETUAL_WIGGLE_DELAY));
+
+			avgShardBytes.reset();
+			self->getAverageShardBytes.send(avgShardBytes);
+			int64_t avgBytes = wait(avgShardBytes.getFuture());
+			double ratio = self->loadBytesBalanceRatio(avgBytes * SERVER_KNOBS->PERPETUAL_WIGGLE_SMALL_LOAD_RATIO);
+			bool imbalance = ratio < SERVER_KNOBS->PERPETUAL_WIGGLE_MIN_BYTES_BALANCE_RATIO;
+			CODE_PROBE(imbalance, "Perpetual Wiggle pause because cluster is imbalance.");
+
+			// there must not have other teams to place wiggled data
+			takeRest = self->server_info.size() <= self->configuration.storageTeamSize ||
+			           self->machine_info.size() < self->configuration.storageTeamSize || imbalance;
+
+			self->storageWiggler->setWiggleState(StorageWiggler::PAUSE);
```

Comment:
I'm confused how this is functioning correctly?  The while() loop begins, then takeRest is only modified in the line before this one, but then regardless of the new value `setWiggleState(StorageWiggler::PAUSE)` is done, and then the loop will stop if `takeRest` is false as the while condition will now be false, but the wiggle state will be left Paused.  

This function should probably have a comment block explaining what it does and its exit conditions.

Reply:
The wiggle state will be set to`RUN` when the next storage server is start wiggling, which is in `perpetualStorageWiggler`